website/docs/sdk-reference/dotnet.md

```diff
@@ -240,19 +240,66 @@ User userObject = new User("john@example.com");
 | `Id`       | **REQUIRED.** Unique identifier of a user in your application. Can be any `string` value, even an email address.                |
 | `Email`    | Optional parameter for easier targeting rule definitions.                                                                       |
 | `Country`  | Optional parameter for easier targeting rule definitions.                                                                       |
-| `Custom`   | Optional dictionary for custom attributes of a user for advanced targeting rule definitions. e.g. User role, Subscription type. |
+| `Custom`   | Optional dictionary for custom attributes of a user for advanced targeting rule definitions. E.g. User role, Subscription type. |
 
 ```csharp
 User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
 {
     Email = "john@example.com",
     Country = "United Kingdom",
-    Custom = new Dictionary<string, string> {
-        {"SubscriptionType", "Pro"},
-        {"UserRole", "Admin"}}
+    Custom =
+    {
+        ["SubscriptionType"] = "Pro",
+        ["UserRole"] = "Admin"
+    }
+};
+```
+
+The `Custom` dictionary also allows attribute values other than `string` values:
+
+```csharp
+User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
+{
+    Custom =
+    {
+        ["Rating"] = 4.5,
+        ["RegisteredAt"] = DateTimeOffset.Parse("2023-11-22 12:34:56 +00:00", CultureInfo.InvariantCulture),
+        ["Roles"] = new[] { "Role1", "Role2" }
+    }
 };
 ```
 
+The set of allowed attribute values depends on the comparison type of the condition which references the User Object attribute.
+`string` values are supported by all comparison types (in some cases they need to be provided in a specific format though).
+Some of the comparison types work with other types of values, as described below.
+
+**Text-based comparisons** (EQUALS, IS ONE OF, etc.)
```

Comment:
```suggestion
**Text-based comparators** (EQUALS, IS ONE OF, etc.)
```

Again, not sure

Reply:
This text was copied from the SDK intellisense docs, which was approved already, so I wouldn't change this.

Reply:
Same as above then :/

Reply:
Reminder: after discussing this, we decided on the term "comparator".